driver-core/src/main/com/mongodb/internal/connection/ConcurrentLinkedDeque.java

```diff
@@ -901,4 +909,20 @@ public void remove() {
     public Iterator<E> descendingIterator() {
         throw new UnsupportedOperationException();
     }
+
+    public interface RemovalReportingIterator<E> extends Iterator<E> {
+        /**
+         * Removes from the underlying collection the last element returned by this iterator and reports whether the current element was
+         * removed by the call.  This method can be called only once per call to {@link #next}.
+         *
+         * @return true if the element was successfully removed by this call, false if the element had already been removed by a concurrent
+         * removal
+         * @implSpec
```

Comment:
I had to look up `@implSpec`  - two questions 1) should it be empty / have no description. 2) will we build java doc with this enabled?

Reply:
Removed it since it's empty.